lib/workers/repository/init/merge.ts

```diff
@@ -69,7 +70,9 @@ export async function detectRepoFileConfig(): Promise<RepoFileConfig> {
       configFileRaw = null;
     }
     if (configFileRaw) {
-      let configFileParsed = JSON5.parse(configFileRaw);
+      let configFileParsed = configFileName.endsWith('.json5')
+        ? JSON5.parse(configFileRaw)
+        : parseJsonWithFallback(configFileRaw, configFileName);
```

Comment:
I think we should add another `parseJson` function to simplify those checks futher.
This parse can later easily extended to support yaml.

```ts
function parseJson(content: string, filename: string): unknown {
  return filename.endsWith('.json5') ? JSON5.parse(content): parseJsonWithFallback(content, filename);
}